src/conn.rs

```diff
@@ -191,6 +196,8 @@ pub struct InProgress<'a, 'c> {
     partition_map: PartitionMap,
     schema: Schema,
     cache: RwLockWriteGuard<'a, AttributeCacher>,
+    transactions: Option<BatchedTransaction>,
+    observer_service: &'a Mutex<TxObservationService>,
```

Comment:
It would be nice if we simply didn't do any of this work if there are no observers at the point of instantiating the `InProgress`.